src/components/DelegationDashboard/DashboardTrack.tsx

```diff
@@ -106,6 +107,25 @@ const DashboardTrackListing = ({ className, posts, trackDetails, totalCount }: P
 		// eslint-disable-next-line react-hooks/exhaustive-deps
 	}, [network]);
 
+	const handleUndelegationDisable = (item: any) => {
+		if (!item?.length || !item?.[0]?.delegatedOn || !item?.[0]?.lockPeriod) return { delegationDisable: false, timeLeftInUndelegation: { percentage: 0, time: null } };
+
+		const lockedDays = item?.[0]?.lockPeriod ? 7 * 2 ** (item?.[0]?.lockPeriod - 1) : 0;
```

Comment:
please add comments to the code or make constants, there are too many magic numbers in the code.
e.g. Why the 7 * 2 ** and why are we subtracting 1 from the lockPeriod ?